logstash-core/src/main/java/org/logstash/common/FsUtil.java

```diff
@@ -36,6 +41,18 @@ public static boolean hasFreeSpace(final String path, final long size)
             }
         }
 
+        final long freeSpace = location.getFreeSpace();
+
+        if (freeSpace == 0L && IS_WINDOWS) {
```

Comment:
Is it worth adding a check here for Java versions <9 so that the free space test is re-enabled when run on Java versions with the fix for SUBST drives?

Reply:
good idea - but first need to validate that it is indeed fixed in Java 9. Saw a JVM issue about it but did not go as far as testing it on Java 9.

Reply:
@danhermann I just rana test on Java 9 and free space is still reported a `0L` on SUBST'ed drive so I think we should move forward, and I'll create a followup issue to see if we need to add and alternate free space strategy on Windows.

Reply:
Cool, sounds good to me